pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -63,6 +64,24 @@ func (src *EnvironmentResource) ConvertTo() (v1.DataModelInterface, error) {
 	}
 	converted.Properties.Compute = *envCompute
 
+	recipeConfig := src.Properties.RecipeConfig
+	if recipeConfig != nil {
+		if recipeConfig.Terraform != nil {
+			if recipeConfig.Terraform.Authentication != nil {
```

Comment:
Can these conditions be collapsed? Or do we expect that there may be future logic that's unique to each config level?

Reply:
In any case i think we still need to check nil at every level to avoid null pointer. Let me know if you think otherwise